R/motif_enrichment_monaLisa.R

```diff
@@ -348,7 +354,28 @@
     df$GCwgt <- ifelse(df$isForeground,
                        1.0,
                        weight_per_bin[as.character(df$GCbin)])
-
+    
+    # adjust for sequence lengths (per GC bin)
+    if (normalizeByLength) {
+        # ... get sequence lengths
+        seq_lengths <- width(df$seqs)
+        # ... get median length of foreground sequences per GC bin
+        median_FG_length_per_GCbin <- vapply(
+            X = as.character(names(weight_per_bin)), 
+            FUN = function(x){
+                median(seq_lengths[df$isForeground & as.character(df$GCbin)==x])
+            }, 
+            FUN.VALUE = 0)
+        median_FG_length_per_GCbin_vector <- 
+            ifelse(df$isForeground,
+                   NA,
+                   median_FG_length_per_GCbin[as.character(df$GCbin)])
+        # ... correct background sequence weights
+        w_bg <- !df$isForeground
+        df$GCwgt[w_bg] <- df$GCwgt[w_bg] * seq_lengths[w_bg] / median_FG_length_per_GCbin_vector[w_bg]
```

Comment:
Dania's tests show that the length correction does the right thing, my intuition is confused, though. For example, if a bg seq is twice as long as the median fg seq in that bin, it's weight will be doubled (not halfed). Somehow I was assuming that longer sequences should be downweighted relative to shorter ones??